pkg/kubelet/kubelet.go

```diff
@@ -287,24 +287,16 @@ func makePodSourceConfig(kubeCfg *kubeletconfiginternal.KubeletConfiguration, ku
 }
 
 // PreInitRuntimeService will init runtime service before RunKubelet.
-func PreInitRuntimeService(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
-	kubeDeps *Dependencies,
-	remoteRuntimeEndpoint string,
-	remoteImageEndpoint string) error {
-	// remoteImageEndpoint is same as remoteRuntimeEndpoint if not explicitly specified
-	if remoteRuntimeEndpoint != "" && remoteImageEndpoint == "" {
-		remoteImageEndpoint = remoteRuntimeEndpoint
-	}
```

Comment:
moving this cross-field defaulting from here to API defaulting means a user-specified `--container-runtime-endpoint` flag will leave ImageServiceEndpoint with an incorrect default value

Reply:
Maybe now we should leave it here and the API default set only default `remoteRuntimeEndpoint` or set nothing in API default.
Once we deprecated the flag and remove it, then we can move from here to API defaulting at that time.

Reply:
Yes. It makes sense to leave this assignment here (and not in defaults.go) and to have a default set in defaults.go for only the `ContainerRuntimeEndpoint`.